Docs: Updates partners documentation to support WooCommerce extension bundles

oikeme opened the pull request:
Update the Readme.md to include a link to the plan-provisoning-woocommerce.md which has been newly added. This document will be used for information on variances specific to provisioning WooCommerce extensions through the Jetpack hosting partner program.

Fixes N/A

#### Changes proposed in this Pull Request:
No functional changes. Updates to the main Readme.md and adding a new document.

#### Testing instructions:
N/A

#### Proposed changelog entry for your changes:
No changelog entry needed.

oikeme committed:
Update Documentation

Update the Readme.md to include a link to the plan-provisoning-woocommerce.md which has been newly added. This document will be used for information on variances when provisioning WooCommerce plans.

ebinnion commented on docs/partners/README.md:
```diff
@@ -18,7 +18,9 @@ If you have any technical questions or concerns, don’t hesitate to get in touc
   - [Provisioning status](determining-provisioning-status.md)
 - [Managing modules](managing-modules.md)
 - [Upgrade redirection](upgrade-redirection.md)
+- [Woo Commerce Provisioning](plan-provisioning-woocommerce.md)
```
WooCommerce is a single word

ebinnion commented on docs/partners/plan-provisioning-woocommerce.md:
```diff
@@ -0,0 +1,10 @@
+# Provisioning Woo Commerce Extensions
+If your Jetpack hosting partner account supports creating WooCommerce Extensions through the Jetpack hosting partner program this document provides additional information.
```
I'd probably add a command before "this document provides additional information".

ebinnion commented on docs/partners/plan-provisioning-woocommerce.md:
```diff
@@ -0,0 +1,10 @@
+# Provisioning Woo Commerce Extensions
+If your Jetpack hosting partner account supports creating WooCommerce Extensions through the Jetpack hosting partner program this document provides additional information.
+
+All other documentation provided for the Jetpack partner hosting program is the same and valid for WooCommerce extension provisioning however there are a few extra details for provisioning extensions which are outlined below.
```
I'd probably add a command before "however".

ebinnion commented on docs/partners/plan-provisioning-woocommerce.md:
```diff
@@ -0,0 +1,10 @@
+
+All other documentation provided for the Jetpack partner hosting program is the same and valid for WooCommerce extension provisioning however there are a few extra details for provisioning extensions which are outlined below.
+
+## Provisioning Plan
```
This section seems like a bit much. Perhaps we can summarize it and combine it with the above? Maybe something like:

> All other documentation provided for the Jetpack partner hosting program is the same and valid for WooCommerce extension provisioning, except the `plan` argument will need to change. At the moment, WooCommerce extension bundles are hosting partner specific, so please get in touch with us to get your plan slug if you do not already have it.

oikeme committed:
Documentation updates

Removed the provisioning plan section and combined it with the first section of the documentation including the link to the standard provisioning documentation. Also fixed a typo with WooCommerce in the main Readme.md

jeherve commented:
Is this ready for review?

ebinnion commented:
I'm going to close this since we're going to take another approach for provisioning WooCommerce extensions.
